app/src/main/java/com/worldsnas/starplayer/view/PlayerFragment.kt

```diff
@@ -12,19 +11,23 @@ import com.google.android.exoplayer2.source.MediaSource
 import com.google.android.exoplayer2.source.ProgressiveMediaSource
 import com.google.android.exoplayer2.upstream.DefaultDataSourceFactory
 import com.worldsnas.starplayer.App
+import com.worldsnas.starplayer.ConstValues
 import com.worldsnas.starplayer.databinding.FragmentPlayerBinding
 import com.worldsnas.starplayer.di.DaggerPlayerComponent
 import com.worldsnas.starplayer.di.PlayerComponent
-import java.io.File
+import com.worldsnas.starplayer.model.Music
 
 /**
- * A simple [Fragment] subclass.
+ * Fragment to Play Musics
  */
 class PlayerFragment : Fragment() {
 
-    lateinit var playerComponent: PlayerComponent
-    var exoPlayer: SimpleExoPlayer? = null
-    var viewBinding: FragmentPlayerBinding? = null
+    private lateinit var playerComponent: PlayerComponent
+    private var exoPlayer: SimpleExoPlayer? = null
+    private var viewBinding: FragmentPlayerBinding? = null
+    private val music by lazy {
```

Comment:
benazaram ye `lateinit var` sade kheili behtare
lazy overhead e ziadi dare

Reply:
typesh ham nullable nabashe kheili behtare
mantegh:
in class bedone in object nemitone kar kone 
pas age kasi in object ro yadesh raft befreste ma `throw` konim kheili rahat mishe control va peyda kard moshkel ro

Reply:
try catch khobe?